src/transformations.jl

```diff
@@ -72,14 +74,16 @@ end
 
 
 """
-    smooth_node(node::StructLogicCircuit, parent_scope, scope, lit_nodes)
+    smooth_node(node::StructLogicCircuit, parent_scope, scope, lit_nodes, vtree_leaves)
 
 Return a smooth version of the node where 
 the are added to the scope by filling the gap in vtrees, using literals from `lit_nodes`
 """
-function smooth_node(node::StructLogicCircuit, parent_scope, scope, lit_nodes)
+function smooth_node(node::StructLogicCircuit, parent_scope, scope, lit_nodes, vtree_leaves)
     # Compute the vtrees based on variables used
-    target_vtr = lca(map(x -> vtree(lit_nodes[x]), collect(parent_scope))...)
+    target_vtr = lca(map(collect(parent_scope)) do x
+                         vtree(get!(lit_nodes, x, PlainStructLiteralNode(Lit(x), vtree_leaves[Var(x)])))
```

Comment:
Here we are only adding positive literals, does that also handle the case if a negative literal is misisng from the original circuit?

Reply:
You are right, which means I did not do a very good job at testing. :) Will fix and add a new test covering that case.

Reply:
Upon further reflection, I don't think we do need to cover the case of `¬x`, since we only need to find the `lca` of the vtree on this line. The disjoining done in the rest of the recursion should take care of the rest. Here's a simple case where this happens:

```julia
# Create a (1, 2) vtree
V = Vtree(PlainVtreeLeafNode.([1, 2])...)
# Literals
x = PlainStructLiteralNode(Lit(1), V.left)
y = PlainStructLiteralNode(Lit(2), V.right)
# A simple (1 ∧ 2) ∨ 1 where no negative literals show up
L = PlainStruct⋁Node([PlainStruct⋀Node(x, y, V), x], V)
# Plot:
using TikzPictures; TikzPictures.save(PDF("/tmp/a.pdf"), LogicCircuits.plot(L))
```
<img width="56" alt="a" src="https://user-images.githubusercontent.com/3943662/154344866-58938e5f-e70f-45ee-a4d1-37e1085f8ad4.png">

```julia
# Should output the smooth circuit (1 ∧ 2) ∨ (1 ∧(2 ∨ ¬2))
L_s = smooth(L)
# Works fine
TikzPictures.save(PDF("/tmp/b.pdf"), LogicCircuits.plot(L))
```

<img width="86" alt="b" src="https://user-images.githubusercontent.com/3943662/154344919-13650862-619a-4de0-a953-2f816f7e0495.png">

Reply:
Interesting, thanks for the detailed example. I guess I will merge now then.

Reply:
I will double check with script from https://github.com/Juice-jl/ProbabilisticCircuits.jl/issues/80 and then merge.